VotingApplication/VotingApplication.Web/Views/Routes/ManageOptions.cshtml

```diff
@@ -4,23 +4,27 @@
-            <div>
-                <span class="glyphicon glyphicon-remove pull-right clickable" aria-hidden="true" ng-click="remove(option)"></span>
-                <span class="glyphicon glyphicon-pencil pull-left clickable" ng-if="!editToggle" aria-hidden="true" ng-click="$parent.editToggle=true"></span>
-                <span class="glyphicon glyphicon-ok pull-left clickable" ng-if="editToggle" aria-hidden="true" ng-click="editToggle=false;updatePoll()"></span>
-                <h4 class="icon-padded-left" ng-if="!editToggle">{{option.Name}}</h4>
-                <h4 class="icon-padded-left" ng-if="editToggle"><input ng-model="option.Name" /></h4>
-            </div>
-            <div>
-                <span class="icon-padded-left" ng-if="!editToggle">{{option.Description}}</span>
-                <span class="icon-padded-left" ng-if="editToggle"><input ng-model="option.Description" /></span>
-            </div>
+            <form ng-submit="editToggle=false;updatePoll()">
+                <div>
+                    <span class="glyphicon glyphicon-remove pull-right clickable" aria-hidden="true" ng-click="remove(option)"></span>
+                    <span class="glyphicon glyphicon-pencil pull-left clickable" ng-if="!editToggle" aria-hidden="true" ng-click="$parent.editToggle=true"></span>
+                    <span class="glyphicon glyphicon-ok pull-left clickable" ng-if="editToggle" aria-hidden="true" ng-click="editToggle=false;updatePoll()"></span>
+                    <h4 class="icon-padded-left" ng-if="!editToggle">{{option.Name}}</h4>
+                    <h4 class="icon-padded-left" ng-if="editToggle"><input ng-model="option.Name" autofocus /></h4>
+                </div>
+                <div>
+                    <span class="icon-padded-left" ng-if="!editToggle">{{option.Description}}</span>
+                    <span class="icon-padded-left" ng-if="editToggle"><input ng-model="option.Description" /></span>
+                    @*Angular requires a button for submission. Go figure.*@
+                    <button type="submit" hidden></button>
+                </div>
+            </form>
         </div>
         <div class="manage-section">
-            <form name="newOptionForm" ng-submit="add(newOptionForm);clear(newOptionForm)">
+            <form name="newOptionForm" ng-submit="add(newOptionForm);clear(newOptionForm);updatePoll();">
                 <div>
-                    <span class="glyphicon glyphicon-plus pull-right clickable" aria-hidden="true" ng-hide="newOptionForm.$invalid" ng-click="add(newOptionForm);clear(newOptionForm)"></span>
-                    <h4 class="icon-padded-left"><input ng-model="newOptionForm.Name" placeholder="Option Name" required /></h4>
+                    <span class="glyphicon glyphicon-plus pull-right clickable" aria-hidden="true" ng-hide="newOptionForm.$invalid" ng-click="add(newOptionForm);clear(newOptionForm);updatePoll();"></span>
```

Comment:
The add/update buttons shouldn't be calling updatePoll, only the "Save and Back" should be